### PR TITLE
Match exact set of errors in schema validation tests

### PR DESCRIPTION
**Pull request by IvanGoncharov**
> Ideally these should detect the exact set of schema validation errors and we should write the test so that each test introduces only the validation error it expects to catch

@leebyron Done

### CODE REVIEW

File: src/type/__tests__/validation-test.js [hunk 1]
```diff
@@ -991,21 +1079,47 @@ describe('Type System: Interface fields must have output types', () => {
...
   });
 
+  it('rejects a non-type value as an Interface field type', () => {
+    const schema = schemaWithInterfaceFieldOfType(Number);
+    expect(validateSchema(schema)).to.deep.equal([
+      {
+        message: `The type of BadInterface.badField must be Output Type but got: ${Number}.`,
+      },
+      {
+        message: `Expected GraphQL named type but got: ${Number}.`,
```
Comment (leebyron) [inline src/type/__tests__/validation-test.js]: This one feels low value and duplicative of the error below it, don't you think?

File: src/type/__tests__/validation-test.js [hunk 2]
```diff
@@ -1078,21 +1205,37 @@ describe('Type System: Field arguments must have input types', () => {
...
   });
 
+  it('rejects a non-type value as a field arg type', () => {
+    const schema = schemaWithArgOfType(Number);
+    expect(validateSchema(schema)).to.deep.equal([
+      {
+        message: `The type of BadObject.badField(badArg:) must be Input Type but got: ${Number}.`,
+      },
+      {
+        message: `Expected GraphQL named type but got: ${Number}.`,
```
Comment (leebyron) [inline src/type/__tests__/validation-test.js]: Another example. This is an opportunity to improve the quality of these errors in the future

File: src/type/__tests__/validation-test.js [hunk 3]
```diff
@@ -1144,21 +1289,37 @@ describe('Type System: Input Object fields must have input types', () => {
...
   });
 
+  it('rejects a non-type value as an input field type', () => {
+    const schema = schemaWithInputFieldOfType(Number);
+    expect(validateSchema(schema)).to.deep.equal([
+      {
+        message: `The type of BadInputObject.badField must be Input Type but got: ${Number}.`,
+      },
+      {
+        message: `Expected GraphQL named type but got: ${Number}.`,
```
Comment (leebyron) [inline src/type/__tests__/validation-test.js]: Same

File: src/type/__tests__/validation-test.js (hunk 1 above)
Comment (IvanGoncharov) [reply src/type/__tests__/validation-test.js]: @leebyron Agree 👍 
However I don't see how `validate` can distinguish beetween those two cases:
```js
validateSchema(new GraphQLSchema({
  queryType: Number,
}));
```
```js
validateSchema(new GraphQLSchema({
  queryType: new GraphQLObjectType({
    name: 'Foo',
    fields: { bar: { type: GraphQLString } },
  }),
  types: [Number],
}));
```

In the first case `Expected GraphQL named type but got: ${Number}.` is redundant but in second one it's the only error that would be reported.
Can you suggest how to solve this?

**Pull request merged**